x/distribution/spec/05_hooks.md

```diff
@@ -5,26 +5,44 @@ order: 5
 # Hooks
 
 ## Create or modify delegation distribution
- 
+
  - triggered-by: `staking.MsgDelegate`, `staking.MsgBeginRedelegate`, `staking.MsgUndelegate`
 
-The pool of a new delegator bond will be 0 for the height at which the bond was
-added, or the withdrawal has taken place. This is achieved by setting
-`DelegationDistInfo.WithdrawalHeight` to the height of the triggering transaction. 
+### Before
+
+The delegation rewards are withdrawn to the delegator's withdraw address.
+The rewards include the current period, but excludes the starting period.
+Then increment the validator period, which is used for calculating rewards.
+The latter is necessary because, due to the nature of the triggers, the validator's power and share distribution can have changed.
+The reference count for the delegator's starting period is decremented.
+
+### After
+
+The starting height of the delegation is set to the previous period.
+Because of the `Before`-hook, this is the last period for which the delegator was rewarded.
+
+## Validator created
+
+- triggered-by: `staking.MsgCreateValidator`
+
+Initialized the validator variables: Historical rewards, current accumulated rewards, accumulated commission, and total outstanding rewards.
+All are set to a `0` by default.
+The period is set to `1`.
+
+## Validator removed
+
+- triggered-by: `staking.RemoveValidator`
 
-## Commission rate change
- 
- - triggered-by: `staking.MsgEditValidator`
+Any outstanding commission is sent to the validator's self-delegation's withdrawal address.
+Any remaining rewards still stored in the validator gets sent to the community fee pool.
 
-If a validator changes its commission rate, all commission on fees must be
-simultaneously withdrawn using the transaction `TxWithdrawValidator`.
-Additionally the change and associated height must be recorded in a
-`ValidatorUpdate` state record.
+Note: the validator only gets removed when it has no remaining delegations.
+All outstanding rewards have been withdrawn.
```

Comment:
```suggestion
A validator is removed only after:
- The validator has no more delegations.
- All delegations have ended.
- All outstanding rewards are withdrawn.```

Reply:
Not exactly what I'm going for. It's more of an invariant, that the when the validator is removed, we know that these things have occured. To me the above reads as causation.
I have updated the text with another suggestion, let me know what you think